pandas/tests/io/test_sql.py

```diff
@@ -253,9 +253,13 @@ def _get_exec(self):
         else:
             return self.conn.cursor()
 
-    def _load_iris_data(self, datapath):
+    @pytest.fixture(params=[('io', 'data', 'iris.csv')])
```

Comment:
Is there a technical reason the path needs to passed here as a param (instead of how it was before)? 
We use `datapath` like this in many other places in the tests

Reply:
Hi @jorisvandenbossche yes there is, in all the other places we pass in `datapath` we don't call it directly as a function. Like this code originally did `datapath('io', 'data', 'iris.csv')`. Which is now deprecated, see this [pytest issue](https://github.com/pytest-dev/pytest/issues/3661),  with the suggested refactor.

Reply:
What do you mean with "we don't call it directly as a function" (the `datapath` fixture itself, or the function that it is passed to?).  Because the fixture `datapath` itself is called directly in many other places in the code base.

Reply:
@alimcmaster1 can you respond here?

I'm in the middle of a PR fixing transitioning warnings to errors. Going to skip sql for now, but this will need to be merged first.

Reply:
But, if I go from memory, is it because it's passed as a fixture to the caller of this method? The classes  `setup_method`, which then passes it through to `load_iris_data` as a function, not a fixture?

Reply:
Firstly apologies for the delayed response I havn't had internet access for the past few days. Yes @jorisvandenbossche agree we do, I worded that badly, but I basically meant what @TomAugspurger said above, we want to avoid passing it like `self._load_iris_data(datapath)`. Then calling datapath directly. 

But I see your original point and you are quite right we could write as so
```
@pytest.fixture()
def load_iris_data(self, datapath)
     csv = datapath('io', 'data', 'iris.csv')
```

The code in setup_method would remain as i've currently implemented. But I thought why not parameterize? In case we have another data-set we want to run these tests against in the future?